src/mrack/providers/beaker.py

```diff
@@ -250,7 +249,11 @@ async def wait_till_provisioned(self, resource):
         attempts = 0
         prev_status = ""
 
-        while attempts < self.max_attempts:
+        # let us use timeout variable which is in minutes to define
+        # maximum attempts of polling the beaker hub for job status
+        # with self.poll_sleep which defaults to 30s we should multiply
+        # this number with 2 to get the number of retries
```

Comment:
This may cause troubles in the future. This commit is changing everything from `max_attemps` into `timeout`, but in the end the code is still counting the number of attempts in a not clear way.

Reply:
What is your suggestion?

Reply:
I agree with Armando. It may be very easy to cause problem in future if somebody just changes e.g. wait time without the "magic normalization".  I'd suggest to not mix units in one comparison. E.g. stop using attempts altogether and compare the time from the start of job with timeout.

For time-out case, you can use `else` block of `while else` construct. Instead of the current block which starts with `if attempts >= self.max_attempts:`.

Reply:
I repushed and used timeout.